packages/cluster-ui/src/statementsTable/statementsTable.tsx

```diff
@@ -250,5 +268,39 @@ export function makeNodesColumns(
     },
   ];
 
-  return original.concat(makeCommonColumns(statements, totalWorkload));
+  return original.concat(
+    makeCommonColumns(statements, totalWorkload, nodeRegions),
+  );
+}
+
+// For each statement, generate the list of regions and nodes it was
+// executed on.
+// E.g. of one element of the list: `gcp-us-east1 (n1, n2, n3)`
+export function populateRegionNodeForStatements(
+  statements: AggregateStatistics[],
+  nodeRegions: { [p: string]: string },
+) {
+  statements.forEach(stmt => {
+    const regions: { [region: string]: Set<number> } = {};
+    stmt.stats.nodes.forEach(node => {
+      if (Object.keys(regions).includes(nodeRegions[node.toString()])) {
+        regions[nodeRegions[node.toString()]].add(longToInt(node));
+      } else {
+        regions[nodeRegions[node.toString()]] = new Set([longToInt(node)]);
+      }
+    });
```

Comment:
A short comment summary of this block would be helpful as well.

Reply:
done